test/e2e/quadlet_test.go

```diff
@@ -1075,6 +1075,7 @@ BOGUS=foo
 		Entry("Container - Mount overriding service name", "mount.servicename.container", []string{"service-name.volume"}),
 		Entry("Container - Quadlet Network overriding service name", "network.quadlet.servicename.container", []string{"service-name.network"}),
 		Entry("Container - Quadlet Volume overriding service name", "volume.servicename.container", []string{"service-name.volume"}),
+		Entry("Container - Reuse another container's network", "network.reuse.container", []string{"basic.container"}),
```

Comment:
Because container name can be either set or generated, please add a test case in which the tested unit references a `.container` unit in which the name of the container is set (instead of `basic.container`)